Purchases/Networking/RCHTTPClient.m

```diff
@@ -107,7 +96,10 @@ - (NSMutableURLRequest *)createRequestWithMethod:(NSString *)HTTPMethod
                                             path:(NSString *)path
                                      requestBody:(NSDictionary *)requestBody
                                          headers:(NSMutableDictionary *)defaultHeaders {
-    NSString *urlString = [NSString stringWithFormat:@"https://%@/v1%@", self.class.serverHostName, path];
+    NSString *urlString = [NSString stringWithFormat:@"%@://%@/v1%@",
```

Comment:
also moved the protocol, mostly so that it's easier to find when you want to override it for local khepri.

Reply:
it could be moved into the proxyURL, but I actually wanted to discourage anyone from not using https (the only real reason to use http is when using a local khepri)

Reply:
I had a similar thought when doing Android. I thought that someone could pass a proxy URL starting with `https://` and the base url will end up like `https://https://`. That's why I ended up letting them set a `URL` instead of a String. But as you say they can set it up as `http` which is bad.

In summary, if I want to do something like this in Android, where they set a string that looks like `aproxy.com`, and we always set it to `https`, we lose the validation that making them set a `URL` gives. The `URL` constructor will check the URL is well constructed.

Reply:
updated to NSURL!